### PR TITLE
CONTRIBUTING.md is introduced

### PR DESCRIPTION
**Pull request by AhmetCanSolak**
With this PR, I am introducing a contributing guideline. I tried to keep it simple and I am planning to add more as we discussed in an issue possibly. This PR is created to initialize such guide and provoke us to discuss on this matter.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Contributing to napari-gui
+
+We welcome your contributions! Please see the provided steps below and never hesitate to contact us.
+
+## Steps to contribute
+
+- Ensure that your contribution is in its own branch in your fork of the repository (no other changes should be in the branch).
```
Comment (kne42) [inline CONTRIBUTING.md]: Could you include git instructions with this?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+- Each commit in your branch that you are submitting must have your email address as the 'author'. If you need to configure this specifically for your clone of the repository (because you also work on other projects using git), use git config user.email <address> in that repository clone to set the address for that single clone.
+- Use git rebase to ensure that your contribution applies cleanly to the current HEAD of the master branch in the repository. This also provides a good opportunity to 'squash' any commits in your branch that you'd rather not have live on in infamy!
```
Comment (kne42) [inline CONTRIBUTING.md:27]: Again, I think that git instructions would be helpful here.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+- Create a GitHub 'pull request' for your branch, targeted at master.
```
Comment (kne42) [inline CONTRIBUTING.md]: How?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Contributing to napari-gui
+
+We welcome your contributions! Please see the provided steps below and never hesitate to contact us.
+
+## Steps to contribute
```
Comment (kne42) [inline CONTRIBUTING.md:5]: We should include some basic instructions on how to install the development version, get started with git, etc. here.

**Review by kne42 — CHANGES_REQUESTED** [review]
It looks good but I'd like to see more detailed instructions pertaining to git/GitHub functionality. We should keep in mind that some contributors may not be familiar with git, especially the more complex things like rebasing.

File: CONTRIBUTING.md (hunk 4 above)
Comment (AhmetCanSolak) [reply CONTRIBUTING.md:5]: 👍 will be adding to next changes

**Comment by AhmetCanSolak** [comment]
I updated the commit @kne42

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+#### First, make sure you get development version of the repository and create your own branch
+```
+git clone https://github.com/<username>/napari-gui # get the repo
+git checkout dev # checkout to dev branch
+git branch <branchname> # create your own branch
+```
+
+#### Checkout to your own branch for development
+- Ensure that your contribution is in its own branch in your fork of the repository (no other changes should be in the branch). You can it with:
```
Comment (kne42) [inline CONTRIBUTING.md]: "You can do it with"

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+We welcome your contributions! Please see the provided steps below and never hesitate to contact us.
+
+## Steps to contribute
+
+#### First, make sure you get development version of the repository and create your own branch
+```
+git clone https://github.com/<username>/napari-gui # get the repo
+git checkout dev # checkout to dev branch
+git branch <branchname> # create your own branch
```
Comment (kne42) [inline CONTRIBUTING.md:13]: maybe even just `git checkout dev -b <branchname>` or is this too confusing?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+#### Checkout to your own branch for development
+- Ensure that your contribution is in its own branch in your fork of the repository (no other changes should be in the branch). You can it with:
+```
+git checkout BRANCHNAME
+git status
+```
```
Comment (kne42) [inline CONTRIBUTING.md:21]: Maybe include a section about updating the dev branch via `git pull upstream dev` and the current working branch via `git fetch upstream dev; git rebase -Xtheirs upstream/dev` (will need to verify I'm not sure if this is right)

**Review by kne42 — APPROVED** [review]

**Pull request merged**